public/js/angular/directives/wysiwyg.js

```diff
@@ -0,0 +1,405 @@
+        }
+
+        scope.toggleFullscreen = function() {
+          scope.wysiwyg.fullscreen = !scope.wysiwyg.fullscreen;
+
+          if(scope.wysiwyg.fullscreen) {
+            angular.element(element).css({
+              'background-color': '#FFFFFF',
+              'position': 'fixed',
+              'top': '0',
+              'left': '0',
+              'width': '100%',
+              'height': '100%',
+              'overflow': 'auto',
+              'z-index': '10000'
+            }).focus();
+
+            angular.element(element).find('.content_layout').css({
+              'height': (angular.element(element).height() - angular.element(element).find('.content_layout').position().top) + 'px',
+              'margin': '0'
+            });
+          }
+          else {
+            angular.element(element).attr('style', '');
+            angular.element(element).find('.content_layout').attr('style', '');
+          }
+        };
+
+        scope.setPublicLayout = function() {
+          if(!scope.wysiwyg.layout.length) {
+            scope.layout = '';
+            return;
+          }
+
+          if(scope.wysiwyg.currentView === 'html') {
+            scope.wysiwyg.layout = scope.layout.toString();
+            return;
+          }
+
+          var self = this;
+
+          var tempEditable = angular.element(element).find('.temp_editable');
+          tempEditable.html(scope.wysiwyg.layout.toString());
+
+          this.convertReadMore = function() {
+            var s = 0;
+            var readMoreCount = tempEditable.find('.read_more_break').length;
+
+            if(readMoreCount === 0) {
+              scope.layout = tempEditable.html();
+              return;
+            }
+
+            tempEditable.find('.read_more_break').each(function() {
+              if(s === 0) {
+                angular.element(this).replaceWith('^read_more^');
+              }
+              else {
+                angular.element(this).replaceWith('');
+              }
+
+              s++;
+              if(s >= readMoreCount) {
+                scope.layout = tempEditable.html();
+              }
+            });
+          };
+
+          this.convertMedia = function() {
+            var i = 0;
+            var mediaCount = tempEditable.find('.media_preview').length;
+            if(mediaCount === 0) {
+              this.convertReadMore();
+              return;
+            }
+
+            tempEditable.find('.media_preview').each(function() {
+              var mediaTags = ['^' + angular.element(this).attr('media-tag') + '^'];
+              var subTags = angular.element(this).find('[media-tag]');
+              for(var j = 0; j < subTags.length; j++) {
+                mediaTags.push('^' + $(subTags[j]).attr('media-tag') + '^')
+              }
+
+              angular.element(this).replaceWith(mediaTags.concat(''));
+
+              i++;
+              if(i >= mediaCount){
+                self.convertReadMore();
+              }
```

Comment:
Is the .each function synchronous or asynchronous?  If synchronous (I think it is) then move then move the call to ```convertReadMore``` to below the conclusion of the call to ```.each```